packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -552,6 +566,13 @@ function createArrayNodeProxy(
 	allowAdditionalProperties: boolean,
 	customTargetObject?: object,
 ): TreeArrayNode {
+	if (customTargetObject !== undefined) {
+		// This causes `JSON.stringify` to give the same output as if the proxy were truly an array.
+		// Otherwise, it will stringify like an object (e.g. `{"0": ..., "1": ...}`) for a `customTargetObject`.
+		Reflect.defineProperty(customTargetObject, "toJSON", {
+			value: () => Array.from(proxy as any),
```

Comment:
the TreeArrayNode implements iterable, so we shouldn't need a cast here, and certainly don't need `any`